integration/mediation-tests/tests-platform/tests-rabbitmq/src/test/java/org/wso2/carbon/esb/rabbitmq/transport/test/RabbitMQConsumerTestCase.java

```diff
@@ -38,77 +41,67 @@
     public void init() throws Exception {
         super.init();
         initRabbitMQBroker();
+        //The consumer proxy cannot be pre-deployed since the queue declaration(which is done in 'initRabbitMQBroker')
+        // must happen before deployment.
         loadESBConfigurationFromClasspath("/artifacts/ESB/rabbitmq/transport/rabbitmq_consumer_proxy.xml");
         logViewer = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
     }
 
-    private void initRabbitMQBroker() {
-        sender = new RabbitMQProducerClient("localhost", 5672, "guest", "guest");
+    private void initRabbitMQBroker() throws URISyntaxException {
```

Comment:
let's move private functions to the end of the class and provide public functions priority